src/main/scala/za/co/absa/ultet/util/ResourceReader.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.ultet.util
+
+import scala.io.Source
+
+object ResourceReader {
```

Comment:
This approach of reading a file is valid and correct, but I would suggest using scala.util.Using :-) as it handles the closing of given resource automatically. See the snippet below. Really simple and much more concise.
```scala
val fileContent: Try[String] = Using(Source.fromFile(filePath)) { source =>
  source.getLines().mkString("\n")
}
```